java/compaction-core/src/main/java/sleeper/compaction/task/CompactionTaskFinishedStatus.java

```diff
@@ -0,0 +1,183 @@
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(finishTime, totalJobs, totalRuntime,
+                totalRecordsRead, totalRecordsWritten, recordsReadPerSecond, recordsWrittenPerSecond);
+    }
+
+    @Override
+    public String toString() {
+        return "CompactionTaskFinishedStatus{" +
+                "finishTime=" + finishTime +
+                ", totalJobs=" + totalJobs +
+                ", totalRuntime=" + totalRuntime +
+                ", totalRecordsRead=" + totalRecordsRead +
+                ", totalRecordsWritten=" + totalRecordsWritten +
+                ", recordsReadPerSecond=" + recordsReadPerSecond +
+                ", recordsWrittenPerSecond=" + recordsWrittenPerSecond +
+                '}';
+    }
+
+
+    public static final class Builder {
+        private Instant finishTime;
+        private int totalJobs;
+        private double totalRuntime;
+        private long totalRecordsRead;
+        private long totalRecordsWritten;
+        private double recordsReadPerSecond;
+        private double recordsWrittenPerSecond;
+
+        private Builder() {
+        }
+
+        public Builder finishTime(Instant finishTime) {
+            this.finishTime = finishTime;
+            return this;
+        }
+
+        public Builder totalJobs(int totalJobs) {
+            this.totalJobs = totalJobs;
+            return this;
+        }
+
+        public Builder totalRuntime(double totalRuntime) {
+            this.totalRuntime = totalRuntime;
+            return this;
+        }
+
+        public Builder totalRecordsRead(long totalRecordsRead) {
+            this.totalRecordsRead = totalRecordsRead;
+            return this;
+        }
+
+        public Builder totalRecordsWritten(long totalRecordsWritten) {
+            this.totalRecordsWritten = totalRecordsWritten;
+            return this;
+        }
+
+        public Builder recordsReadPerSecond(double recordsReadPerSecond) {
+            this.recordsReadPerSecond = recordsReadPerSecond;
+            return this;
+        }
+
+        public Builder recordsWrittenPerSecond(double recordsWrittenPerSecond) {
+            this.recordsWrittenPerSecond = recordsWrittenPerSecond;
+            return this;
+        }
+
+        public Builder addJobStatus(CompactionJobStatus jobStatus) {
+            totalJobs++;
+            totalRecordsRead += jobStatus.getFinishedSummary().getLinesRead();
+            totalRecordsWritten += jobStatus.getFinishedSummary().getLinesWritten();
+            return this;
+        }
+
+        public Builder finish(Instant finishTime, double duration) {
+            recordsReadPerSecond = totalRecordsRead / duration;
+            recordsWrittenPerSecond = totalRecordsWritten / duration;
+            totalRuntime = duration;
+            this.finishTime = finishTime;
+            return this;
+        }
+
+        public CompactionTaskFinishedStatus empty() {
+            return builder().build();
+        }
```

Comment:
This isn't a well-behaved empty value since you would always expect a finished status to have a finished time. We can use null.